sdk/core/core-client-lro-rest/src/getLongRunningHelper.ts

```diff
@@ -0,0 +1,183 @@
+): PollerLike<PollOperationState<TResult>, TResult>;
+/**
+ * Helper function that builds a Poller object to help polling a long running operation.
+ * @param client - Client to use for sending the request to get additional pages.
+ * @param initialResponse - The initial response.
+ * @param pollerOptions - Options to set a resume state or custom polling interval.
+ * @param finalStateVia - Additional information about how to get to the final result of the long running operation.
+ * Typically this is described in the Swagger using the extension `x-ms-long-running-operation-options`.
+ * @returns - A poller object to poll for operation state updates and eventually get the final response.
+ */
+export function getLongRunningPoller<TResult extends HttpResponse>(
+  client: Client,
+  initialResponse: TResult,
+  pollerOptions: LROPollerOptions,
+  finalStateVia: FinalStateVia
+): PollerLike<PollOperationState<TResult>, TResult>;
+export function getLongRunningPoller<TResult extends HttpResponse>(
+  client: Client,
+  initialResponse: TResult,
+  pollerOptionsOrFinalState?: LROPollerOptions | FinalStateVia,
+  finalStateViaOrNothing?: FinalStateVia
+): PollerLike<PollOperationState<TResult>, TResult> {
+  // Default values
+  let pollerOptions: LROPollerOptions = {};
+  let finalStateVia: FinalStateVia | undefined;
+
+  // Figure out which override we are handling
+  if (isFinalStateVia(pollerOptionsOrFinalState)) {
+    finalStateVia = pollerOptionsOrFinalState;
+  } else if (finalStateViaOrNothing) {
+    finalStateVia = finalStateViaOrNothing;
+  }
+
+  if (isLROPollerOptions(pollerOptionsOrFinalState)) {
+    pollerOptions = pollerOptionsOrFinalState;
+  }
+
+  // Translating the rest response into a response that the LRO engine understands
+  const initialLROResponse = getLroResponse(initialResponse);
+
+  const poller: LRO<TResult> = {
+    requestMethod: initialResponse.request.method,
+    requestPath: initialResponse.request.url,
+    sendInitialRequest: async (initializeState) => {
+      // In the case of Rest Clients we are building the LRO poller object from a response that's the reason
+      // we are not triggering the initial request here, just extracting the information from the
+      // response we were provided.
+      initializeState(initialLROResponse.rawResponse, initialLROResponse.flatResponse);
+      return initialLROResponse;
+    },
+    sendPollRequest: async (config, path) => {
+      // This is the callback that is going to be called to poll the service
+      // to get the latest status. We use the client provided and the polling path
+      // which is an opaque URL provided by caller, the service sends this in one of the following headers: operation-location, azure-asyncoperation or location
+      // depending on the lro pattern that the service implements. If non is provided we default to the initial path.
+      const response = (await client
+        .pathUnchecked(path ?? initialResponse.request.url)
```

Comment:
The `path ?? initialResponse.request.url` part here confuses me.  In what cases will `sendPollRequest` be called with a falsy value?  It feels like we should just not allow that to happen.  I would not expect that in all cases the URI of the initial request can be used to fetch the current operation status (I don't think this is correct, for example, for cases where the final state is original-uri and we have an operation-location header in the initial response (since it will point at a different URI for the actual operation status).

Reply:
Good catch! @deyaaeldeen is looking into handling the undefined internally so path is always a `string`

Reply:
@ellismg I opened this draft PR to discuss the design of the lro engine: https://github.com/Azure/azure-sdk-for-js/pull/15949. Please feel free to chime in!

Reply:
`sendPollRequest`'s path arg is now required.